lib/postgresql.js

```diff
@@ -500,9 +524,14 @@ PostgreSQL.prototype._buildWhere = function(model, where) {
     }
     var p = props[key];
     if (p == null) {
-      // Unknown property, ignore it
-      debug('Unknown property %s is skipped for model %s', key, model);
-      continue;
+      if (isNested(key)) {
```

Comment:
would `p` be null if the property has nested json keys? In other words, could we move this condition outside of the `p == null` check? I'm worried that this is unreachable code, unless I'm missing something.

Reply:
`p` can be null if the property is not defined on the model. Model might have `foo` property but `foo.bar` is being queried so `props[key]` is null but `isNested(key)` is true.

On the flip side if we try to query `baz` (which doesn't exist on the model and isn't a nested query) it should fall through so nothing unreachable.

However you do make a point that `p` could be null if it is nested and the property doesn't exist on the model. Will fix that.